### PR TITLE
CFIN-350 Return the number of full matches before the search results array

### PR DESCRIPTION
**Pull request by ncn1**

### CODE REVIEW

**Review by FergusMcGlynn — APPROVED** [review]

File: config/course-search.ftl [hunk 1]
```diff
@@ -9,7 +9,9 @@
     <#-- CALLBACK -->
         <#if RequestParameters.callback?has_content>${RequestParameters.callback}(</#if>
         {
-        "results": [
+        "fullMatches": "${response.resultPacket.resultsSummary.fullyMatching?json_string}",
```
Comment (FergusMcGlynn) [inline config/course-search.ftl:12]: Not sure `fullMatches` is the most intuitive name for number of results. Maybe `numberOfResults` would be clearer?

Comment (ncn1) [reply config/course-search.ftl:12]: I think numberOf Results is confusing, as the following array is called results, and it only contains whatever was specified explicitly in numRanks or 10 by default.

Comment (FergusMcGlynn) [reply config/course-search.ftl:12]: `totalNumberOfResults`? `fullResultCount`? Something to describe the fact that its a number is mainly what I'm getting at. `fullMatches` sounds like "here are the matches".

Comment (ce666-york) [reply config/course-search.ftl:12]: My suggestions are `count` or `totalCount`

**Review by ce666-york — APPROVED** [review]
I think we're going to want to pass back more than this for full pagination functionality - something like Grails' `max` and `offset` so we can work out what the current page is as well as the total number of pages.

File: config/course-search.ftl (hunk 1 above)
Comment (ncn1) [reply config/course-search.ftl:12]: numberOfMatches?

Comment (ChristianBeddows) [reply config/course-search.ftl:12]: I like just having `count` - minimal, and it makes sense in the scope of this response to have `results` and `count` - I don't think people would be confused when they see it

**Review by ChristianBeddows — APPROVED** [review]

**Comment by ncn1** [comment]
> I think we're going to want to pass back more than this for full pagination functionality - something like Grails' `max` and `offset` so we can work out what the current page is as well as the total number of pages.

max and offset are being passed _into_ the Courses API, so this is information that it knows already and has used to make the Funnelback request - therefore it's not necessary to return the same information from Funnelback .

**Review by mrgibbonsuoy — APPROVED** [review]

**Pull request merged**